### PR TITLE
enhance(office-addin): replace React runtime with plain TypeScript

### PR DESCRIPTION
**Pull request by rschlaefli**
## What

Replaces the React and Webpack Office Add-in with a small Rollup-built TypeScript content add-in. The new flow accepts only KlickerUZH evaluation links, embeds them in PowerPoint, and persists each content add-in instance's URL through Office document settings.

## Changes

- builds one runtime bundle with local CSS and removes the production Tailwind CDN
- validates and reconstructs the exact current and legacy KlickerUZH evaluation URL contract, with URI-encoded dynamic values and focused Node tests
- serializes each settings mutation with its `saveAsync` call, and persists rollback state after failed writes
- keeps the iframe on the production Manage origin with `no-referrer` and the minimum required sandbox permissions
- emits production URLs under `/office-addin/` and verifies exact parity between Rollup output and all 13 deployed docs-static files
- removes the obsolete React taskpane, debug launcher, live reload, wrapper scripts, bundle analyzer, and non-runtime artifacts
- aligns the package with TypeScript 6 and the latest compatible Office tooling patches
- excludes the generated deployment HTML copy from Sonar's duplication metric while retaining normal analysis
- refreshes package docs, the engineering wiki, testing guidance, and the [execution plan](https://github.com/uzh-bf/klicker-uzh/blob/NewPPT/project/2026-07-07-pr4643-office-addin-review.md)

## Branch coverage

- Base: `v3` at `c8de9c897`
- Head: `d6f3afb82`
- 45 commits; 51 files; 2,542 insertions and 8,893 deletions
- The long-lived history contains intermediate React, TypeScript, and Rollup iterations that no longer exist at head. Review the current plain-TypeScript runtime and build pipeline.

## Review focus

- Microsoft documents the settings bag as scoped to the content or task-pane add-in instance that created it and the document. The single `embeddedUrl` key therefore remains instance-scoped: https://learn.microsoft.com/en-us/office/dev/add-ins/develop/persisting-add-in-state-and-settings
- `evaluation-url.ts` accepts only the production Manage host, optional `en` or `de`, current `quizzes` or legacy `sessions`, a UUID, `evaluation`, and the backend-compatible lowercase 64-character hexadecimal HMAC
- `settings-mutation-queue.ts` keeps mutation and persistence atomic across handlers, then performs and saves a rollback before allowing the next mutation to begin
- `build:docs` and `verify:docs` must continue to keep the production manifest, runtime files, styles, maps, and assets identical to the built output
- PowerPoint 16.111.1 now proves the current live evaluation workflow, document persistence, instance-scoped settings, and Change URL reset behavior. Legacy migration remains unverified because no legacy deck was available.

## Verification

Full repository checks after the final runtime fix:

- pinned Node 24.16.0 and pnpm 11.5.0 frozen install: passed, including supply-chain policy validation for 3,796 lock entries
- full production build: 22 of 22 tasks passed
- full test build: 20 of 20 tasks passed
- type checks: 24 of 24 tasks passed
- lint, syncpack, AGENTS.md smoke check, Prisma schema sync, and Prisma namespace tests: passed
- Prettier on every tracked branch file and `git diff --check`: passed
- final security review: no high-confidence finding
- final thermo-nuclear maintainability review: passed with no reportable finding
- independent Agy review with Gemini 3.5 Flash High: "Clean. No reportable findings."
- stubbed browser smoke at 1024x768 and 600x400: invalid and valid URLs, embed, fullscreen, Change URL, and zero page errors passed

Current head after the final security, review, and native-verification adjustments:

- Office typecheck, ESLint, three URL tests, production `build:docs`, 13-file deployment parity, and Microsoft manifest validation under pinned Node 24.16.0 and pnpm 11.5.0: passed
- isolated browser smoke of the reconstructed URL, including an encoded HTML-like query value, embed, iframe source, and zero page errors: passed
- PowerPoint 16.111.1 at `bc5d73313`, before the final queue hardening, with the exact candidate bundle and a fresh tunnel-backed manifest:
  - D1 passed: a real evaluation URL validated and rendered its live evaluation chart
  - D2 passed: save, close, and reopen restored the evaluation without another action
  - D3 passed: two add-in instances on separate slides retained different live questions after reopen
  - D5 passed: Change URL cleared one instance and it stayed empty after reopen without affecting the other instance
  - D4 was not run because no legacy deck was available
- exact-commit security and correctness review found one backend-compatibility edge case; the fix and regression test passed its focused follow-up review
- final documentation review fixes pass Prettier and the repository AGENTS.md validation; all four CodeRabbit threads are resolved
- the final persistence hardening passes six focused Office tests, TypeScript, ESLint, production build, Prettier, and 13-file deployment parity locally
- the previous code head `4e81d8bd8` passed all GitHub CI checks, including all eight Playwright shards and the aggregate status gate; the first shard 6 attempt had an unrelated KPRIM assertion failure and its failed-job rerun passed
- current-head GitHub CI passes, including all eight Playwright shards and the aggregate status gate

Warnings and limits:

- host hooks used Node 26 and could not reuse the pinned Node 24 Docker module tree, so the final commits and push bypassed those hooks after the equivalent pinned-Node gates passed
- existing workspace build warnings include Next.js i18n, page-size, Browserslist, and QR-message warnings; none originate in the Office Add-in diff
- the Office-only audit retains two development-tool findings: `adm-zip` in Microsoft's latest manifest validator and `brace-expansion` in the lint tree. Neither package ships in the add-in runtime or processes user-controlled production input. The earlier `uuid` finding is fixed by `office-addin-manifest` 2.1.6.

## Security and privacy

- Office.js remains the only externally loaded runtime script, as required by Microsoft
- the embedded source URL is restricted to the exact validated production origin, and the frame sends no referrer
- no secrets, participant data, exports, evaluation URLs, or private evidence are included
- native screenshots are retained locally rather than uploaded because they show live evaluation content

## Blocking before merge

- [x] current-head GitHub CI passes
- [x] candidate manifest is sideloaded in real PowerPoint; empty state, invalid URL validation, and insertion of two content add-in objects pass
- [x] D1, D2, D3, and D5 pass with a real evaluation URL
- [ ] required maintainer approval

## Follow-up after merge

- validate D4 legacy settings migration when a suitable legacy deck is available
- validate the deployed production manifest and evaluation embed once more in PowerPoint
- toolbar integration and executable packaging remain separate product work

### CODE REVIEW

**Comment by rschlaefli** [comment]
Production-readiness work is complete on the current branch. The production `/office-addin/` paths, local CSS, exact deploy sync, strict URL validation, serialized settings writes, docs, and current Office tooling are in place and verified.

Correction to the original review: Microsoft documents `Office.context.document.settings` as scoped to the content or task-pane add-in instance that created it and the document, so the plain `embeddedUrl` key does not make separate content add-ins overwrite one another: https://learn.microsoft.com/en-us/office/dev/add-ins/develop/persisting-add-in-state-and-settings.

Pinned Node 24 verification passed for the full production build, test build, type checks, lint, formatting, syncpack, Office tests, 13-file deployment parity, and manifest validation. Real multi-instance PowerPoint sideloading remains the explicit manual pre-merge check; post-deployment PowerPoint validation is a separate release check.

File: apps/office-addin/rollup.config.js [hunk 1]
```diff
@@ -1,187 +1,196 @@
...
-  prodUrl = 'https://www.klicker.uzh.ch/office-addin',
-  manifestPaths = ['src/manifest-content.xml', 'src/manifest-taskpane.xml'],
-} = {}) {
+import serve from 'rollup-plugin-serve'
+import { visualizer } from 'rollup-plugin-visualizer'
+
+// Environment variables and constants
+const isDev = process.env.NODE_ENV === 'development'
+const urlDev = 'https://localhost:3020/'
+const urlProd = 'https://www.klicker.uzh.ch/'
```
[the remark this answers is not in the document]
Comment (rschlaefli) [reply apps/office-addin/rollup.config.js]: Fixed on the current branch. Production replacements now use `https://www.klicker.uzh.ch/office-addin/`, `build:docs` deploys the Rollup output, and `verify:docs` confirmed exact parity across all 13 deployed files at `86c7fcc7b`.

File: apps/office-addin/src/content/content.ts [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+let changeEmbeddedUrlButton: HTMLButtonElement | null = null
+let appContainer: HTMLDivElement | null = null
+let urlValidationMessage: HTMLDivElement | null = null
+
+// --- Constants ---
+/**
+ * The key used to store and retrieve the embedded evaluation URL
+ * within the Office document's settings.
+ */
+const SETTINGS_KEY = 'embeddedUrl'
```
[the remark this answers is not in the document]
Comment (rschlaefli) [reply apps/office-addin/src/content/content.ts:19]: Thanks for flagging this. Microsoft documents that the Settings bag is available only to the content or task-pane add-in instance that created it and the document: https://learn.microsoft.com/en-us/office/dev/add-ins/develop/persisting-add-in-state-and-settings. The plain `embeddedUrl` key is therefore instance-scoped, not shared across every embedded content add-in in the deck. We are keeping the simpler key and retaining a real multi-instance PowerPoint sideload as the manual pre-merge acceptance check.

File: apps/office-addin/src/content/content.ts [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+  contentIframe: HTMLIFrameElement
+  embedButton: HTMLButtonElement
+  iframeContainer: HTMLDivElement
+  messageBox: HTMLDivElement
+  messageText: HTMLSpanElement
+  urlInput: HTMLInputElement
+  urlValidationMessage: HTMLDivElement
+}
+
+const SETTINGS_KEY = 'embeddedUrl'
```
[the remark this answers is not in the document]
Comment (rschlaefli) [reply apps/office-addin/src/content/content.ts:19]: Thanks for flagging this. Microsoft documents that the Settings bag is available only to the content or task-pane add-in instance that created it and the document: https://learn.microsoft.com/en-us/office/dev/add-ins/develop/persisting-add-in-state-and-settings. The plain `embeddedUrl` key is therefore instance-scoped, not shared across every embedded content add-in in the deck. We are keeping the simpler key and retaining a real multi-instance PowerPoint sideload as the manual pre-merge acceptance check.

File: .agents/skills/klicker-testing-verification/SKILL.md [hunk 4]
```diff
@@ -9,12 +9,13 @@ Facts about the test landscape: [docs/testing.md](../../../docs/testing.md). Thi
...
+| `packages/graphql` services/schema                              | `pnpm --filter @klicker-uzh/graphql test:local` — one-command bootstrap (real Postgres + Redis + Hatchet); serialized, don't parallelize                                            |
+| UI or user flows                                                | e2e — new specs go to `klicker-playwright-e2e` (primary suite); use `klicker-cypress-e2e` only to keep the frozen legacy suite green                                                |
+| React component appearance/behavior only                        | there is **no component-test layer** — verify in the browser (below) and rely on e2e if a flow covers it                                                                            |
+| Office Add-in source, build, or manifest                        | Run its `check`, `lint`, `build:docs`, `verify:docs`, and `validate` scripts; use a stubbed Office API for browser UI checks and sideload the manifest in PowerPoint before release |
```
[the remark this answers is not in the document]
Comment (rschlaefli) [reply .agents/skills/klicker-testing-verification/SKILL.md]: Fixed in 1549ae80a: the Office Add-in routing row now includes the focused test command.

File: apps/office-addin/README.md [hunk 5]
```diff
@@ -1,55 +1,83 @@
...
+- `src/styles.css` — local styles shipped with the add-in.
+- `rollup.config.js` — build, HTTPS development server, and production asset assembly.
+- `scripts/sync-docs.mjs` — deploy and exact-parity check.
 
-- Note that all Office host applications like Outlook, Word etc. need to be closed when you clear the cache, otherwise clearing the cache is not possible.
-- When developing on the web, don't forget to clear your browser's cache.
-- When you stopped the dev server, wait for a while before you start it again. The waiting is necessary as it takes the server a while to properly stop, even when it seems that the process is already finished. If you don't wait long enough, the dev server won't restart.
+The manifest validator is development-only and processes the reviewed local
+manifest. Its current Microsoft package retains transitive `adm-zip` and `uuid`
+advisories; do not use it on untrusted manifest or archive input.
```
[the remark this answers is not in the document]
Comment (rschlaefli) [reply apps/office-addin/README.md]: Fixed in 1549ae80a: the README now retains only the current adm-zip advisory and the untrusted-input restriction.

File: docs/getting-started.md [hunk 6]
```diff
@@ -16,7 +16,7 @@ tags:
 
 Aligned to Node `24.16.0` and pnpm `11.5.0` across the entire workspace, including the self-contained devcontainer. Pinned in root `package.json`: `volta.node = 24.16.0`, `volta.pnpm = 11.5.0`, `packageManager = pnpm@11.5.0`.
 
-The workspace TypeScript baseline is `~6.0.3` (config-derived from root and package manifests). `apps/office-addin/package.json` intentionally remains on `~5.6.3` outside this baseline; `.syncpackrc.mjs` contains the narrow exception that keeps `pnpm run check:syncpack` honest without forcing that app into this upgrade.
+The workspace TypeScript baseline is `~6.0.3` across all packages, including `apps/office-addin`. The Office Add-in uses the browser/bundler contract (`target: ES2022`, `module: ESNext`, `moduleResolution: Bundler`, `noEmit`) and explicitly loads the `office-js` global types required by TypeScript 6. No syncpack exception is needed.
```
[the remark this answers is not in the document]
Comment (rschlaefli) [reply docs/getting-started.md]: Verified against cypress/package.json and cypress/tsconfig.json: Cypress already uses TypeScript ~6.0.3, so a TypeScript 5 exception would be inaccurate. Commit 1549ae80a instead clarifies that Cypress shares the TypeScript 6 baseline but preserves its legacy non-strict compiler contract, and corrects the stale TypeScript 5 wording in docs/testing.md.
